web/web.go

```diff
@@ -744,6 +739,48 @@ func (h *Handler) status(w http.ResponseWriter, r *http.Request) {
 	h.executeTemplate(w, "status.html", status)
 }
 
+func (h *Handler) runtimeInfo() api_v1.RuntimeInfo {
+	status := api_v1.RuntimeInfo{
+		StartTime:      h.birth,
+		CWD:            h.cwd,
+		GoroutineCount: runtime.NumGoroutine(),
+		GOMAXPROCS:     runtime.GOMAXPROCS(0),
+		GOGC:           os.Getenv("GOGC"),
+		GODEBUG:        os.Getenv("GODEBUG"),
+	}
+
+	if h.options.TSDBCfg.RetentionDuration != 0 {
+		status.StorageRetention = h.options.TSDBCfg.RetentionDuration.String()
+	}
+	if h.options.TSDBCfg.MaxBytes != 0 {
+		if status.StorageRetention != "" {
+			status.StorageRetention = status.StorageRetention + " or "
+		}
+		status.StorageRetention = status.StorageRetention + h.options.TSDBCfg.MaxBytes.String()
+	}
+
+	metrics, err := h.gatherer.Gather()
+	if err != nil {
+		fmt.Println(err)
```

Comment:
We (almost) never want to `fmt.Println()` anything (we use a logging library instead), but in this case we should either return an error to the caller or panic. Since I'm not 100% sure that none of the Prometheus-internal metric collectors ever return errors during gathering, returning an error is probably the way to go to be safe.